src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -654,8 +677,6 @@ int encryptFinal(ByteBuffer src, ByteBuffer dst)
      * @return the number of bytes placed into the <code>out</code> buffer
      */
     int decrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
-        checkDataLength(ibuffer.size(), len);
```

Comment:
Just curious, why remove this checkDataLength(...) call? Seems inconsistent to other methods?

Reply:
All decrypt() does is put the data into the ibuffer.  I didn't find it useful to calculate that data lengths for the GCM rollover when decryption was not happening.  It's probably a personal preference to wait until doFinal().  The situation when ibuffer is > 64M is very unlikely.